lib/ansible/modules/network/nxos/nxos_install_os.py

```diff
@@ -96,125 +115,465 @@
+            # Check mode indicated that ISSU is not possible so issue the
+            # upgrade command without the non-disruptive flag.
+            issu = 'no'
+        commands = build_install_cmd_set(issu, image, kick, 'install')
+        opts = {'ignore_timeout': True}
+        # The system may be busy from the call to check_mode so loop until
+        # it's done.
+        upgrade = check_install_in_progress(module, commands, opts)
+
+        # Special case:  If we encounter a backend processing error at this
+        # stage it means the command was sent and the upgrade was started but
+        # we will need to use the impact data instead of the current install
+        # data.
+        if upgrade['backend_processing_error']:
+            upgrade['upgrade_succeeded'] = True
+            upgrade['use_impact_data'] = True
+
+        if upgrade['use_impact_data']:
+            if upgrade['upgrade_succeeded']:
+                upgrade = impact_data
+                upgrade['upgrade_succeeded'] = True
+            else:
+                upgrade = impact_data
+                upgrade['upgrade_succeeded'] = False
+
+        if not upgrade['upgrade_succeeded']:
+            upgrade['error'] = True
+    return upgrade
 
 
 def main():
     argument_spec = dict(
         system_image_file=dict(required=True),
         kickstart_image_file=dict(required=False),
+        issu=dict(choices=['required', 'desired', 'no', 'yes'], default='no'),
     )
 
     argument_spec.update(nxos_argument_spec)
 
     module = AnsibleModule(argument_spec=argument_spec,
-                                supports_check_mode=True)
+                           supports_check_mode=True)
 
     warnings = list()
     check_args(module, warnings)
 
-    system_image_file = module.params['system_image_file']
-    kickstart_image_file = module.params['kickstart_image_file']
-
-    if kickstart_image_file == 'null':
-        kickstart_image_file = None
-
-    current_boot_options = get_boot_options(module)
-    changed = False
-    if not already_set(current_boot_options,
-                       system_image_file,
-                       kickstart_image_file):
-        changed = True
-
-    install_state = current_boot_options
-    if not module.check_mode and changed is True:
-        set_boot_options(module,
-                         system_image_file,
-                         kickstart=kickstart_image_file)
-
-        if not already_set(install_state,
-                           system_image_file,
-                           kickstart_image_file):
-            module.fail_json(msg='Install not successful',
-                             install_state=install_state)
-
-    module.exit_json(changed=changed, install_state=install_state, warnings=warnings)
+    # This module will error out if the Ansible task timeout value is not
+    # tuned high enough.
+    check_ansible_timer(module)
+
+    global platform
+    platform = get_platform(module)
```

Comment:
This isn't used anywhere?

Reply:
It's not.  I was using it in an earlier version.  I left it because it's extremely likely that it will be used in a future update to handle a corner case that is platform specific.  I would prefer to document this fact in a comment.

Reply:
ACk. Just FYI OS discovery can be done after CLICONF is implemented.